app/lib/service/download_counts/download_counts.dart

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:basics/basics.dart';
+import 'package:json_annotation/json_annotation.dart';
+import 'package:pub_semver/pub_semver.dart';
+part 'download_counts.g.dart';
+
+@JsonSerializable(includeIfNull: false)
+class CountData {
+  // We store at most two years of data, one data point per day.
+  static const countsCapacity = 731;
+  final versionCounts = <({String versionRange, List<int> counts})>[];
+  final patchCounts = <({String versionRange, List<int> counts})>[];
+  final minorCounts = <({String versionRange, List<int> counts})>[];
+  final majorCounts = <({String versionRange, List<int> counts})>[];
+  final totalCounts = <int>[];
+
+  // Last date with processed data.
+  // The date only contains year, month and date.
+  // Hours, minutes and seconds are disregarded.
+  DateTime? lastDate;
```

Comment:
Each `xxxCounts` list contains a list of integers with a count for each day.

If we have `totalCounts = [1, 2, 3]`

Then how many downloads did we have on `lastDate`? 1 or 3?
What date did we have 2 downloads on? is `lastDate - 1` or `lastDate + 1`?

I think it's worth documenting how this works.. maybe with an illustrative example:
```
/// ```
/// totalCounts = [42, 21, 55]
///                 ^   ^   ^____ Download count on lastDate
///                 |   |________ Download count on lastDate - 1 day
///                 |____________ Download count on lastDate - 2 days
```

Or maybe it's the other way around, but an illustration like this would help.

Reply:
Added doc.